Added support for build and test on windows

garyjN7 opened the pull request.

hzoo committed:
update phab links explanation [skip ci]

garyjN7 committed:
Added npm scripts to allow tests and builds to be performed on Windows.

garyjN7 committed:
Added cross-env dev dependency

garyjN7 committed:
Added es6 support for mocha.

kutyel committed:
Add missing description to all package.json files (#4703) [skip ci]

hzoo committed:
add deprecated description for class-constructor-call [skip ci]

hzoo committed:
add deprecated in readme for class-constructor-call [skip ci]

hzoo committed:
add deprecated in readme for class-constructor-call [skip ci]

garyjN7 committed:
Made quote marks consistent.

garyjN7 committed:
Changed scripts in line with makefile changes.

hzoo committed:
fix spacing [skip ci]

garyjN7 committed:
Makefile now calls npm scripts.

chrisprice commented on packages/babel-polyfill/scripts/build-dist.js:
```diff
@@ -0,0 +1,32 @@
+  fs.mkdirSync(distPath);
+}
+
+const browserifyCommand = `${path.join(thisBase, "lib", "index.js")} \
```
For consistency I'd be tempted to extract this expression to a variable

chrisprice commented on packages/babel-runtime/scripts/build-dist.js:
```diff
@@ -3,6 +3,7 @@ var each       = require("lodash/each");
 var fs         = require("fs");
 var _          = require("lodash");
 
+var join       = require("path").join;
```
Unless the changes to this file are critical, I'd be tempted to remove them from this PR to simplify it

chrisprice commented on scripts/run-tests.js:
```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env node
+"use strict";
+
+require("./babel-register.js");
```
Is this required?

garyjN7 replied:
Yes

chrisprice commented on scripts/run-tests.js:
```diff
@@ -0,0 +1,32 @@
+const mocha = new Mocha();
+
+function getTestFiles() {
+  glob.sync(testSearch).forEach(function(file) {
```
Fat arrows?

garyjN7 replied:
?

chrisprice commented on scripts/run-tests.js:
```diff
@@ -0,0 +1,32 @@
+}
+
+function runTests() {
+  testGrep && mocha.grep(testGrep);
```
Personal preference, I prefer if statements. Leave the `&&` to the obfuscator!

garyjN7 replied:
I see short circuit evaluation used all the time and seems quite pedestrian;  i'd hardly call it exotic.

chrisprice commented on scripts/run-tests.js:
```diff
@@ -0,0 +1,32 @@
+function runTests() {
+  testGrep && mocha.grep(testGrep);
+  mocha.ui("tdd").reporter("dot").timeout(10000).run(function(failures) {
+    process.on("exit", function() {
```
How come you wait for `process.on('exit'`?

chrisprice commented:
The diff stats look a bit weird to me, you've added more than you've deleted... Did you forget to delete the old shell scripts or are they still required?

hzoo committed:
update eslint-config, fixes, add commands (#4721)

* update eslint-config, fixes, add commands

* switch plugin

motiz88 committed:
Consolidate contiguous var declarations in destructuring transform (#4690)

* Consolidate contiguous var declarations in destructuring transform

Fixes #3081.

* Simplify var node coalescing in es2015-destructuring

* Revert "Simplify var node coalescing in es2015-destructuring"

This reverts commit 15cb373f0726f68225f7080a7ae206a63af174ee.

* More careful condition for var coalescing in es2015-destructuring

Jamesernator committed:
Fixed incorrect compilation of async iterator methods (#4719)

kaicataldo committed:
Remove babel-doctor from babel-cli (#4725)

* Remove babel-doctor from babel-cli

fixes #4678

* Throw when babel-doctor is used

samwgoldman committed:
Add EmptyTypeAnnotation (#4724)

Added to Flow in facebook/flow#c603505583993aa953904005f91c350f4b65d6bd
Parser support added in babel/babylon#171.

existentialism committed:
Add useBuiltIns option to helper-builder-react-jsx (#4655)

motiz88 committed:
Make special case for class property initializers in `shadow-functions` (#4502)

motiz88 committed:
Update tests for changed error messages in Babylon (#4727)

Depends on babel/babylon#172.

motiz88 committed:
Add .gitattributes forcing LF line endings (#4730)

DrewML committed:
Run ESLint on test files, and fix lint errors in test files (#4732)

DrewML committed:
Change usage of "suite"/"test" in unit-tests to "describe"/"it" (#4734)

Fixes #4733

hzoo committed:
increase git depth [skip ci]

hzoo committed:
contributing updates [skip ci]

existentialism committed:
Fix few typos in issue/pr templates (#4739) [skip ci]

motiz88 committed:
Automatically generate missing expected.js fixtures (#4735)

This is much like Babylon's existing behavior around `expected.json`. The equivalent of babel/babylon#188 is already applied here, to guard against silent failure (and a potential false positive) if a test is accidentally committed without its expected.js.

nhajidin committed:
Fix line endings on checkout

kesne committed:
[import()] Initial support for dynamic-import (#4699)

* [import()] Initial support for function-like import

* [import()] Renaming import-functions to dynamic-import

* [stage-2] Fixing lint error

* add to package.json

existentialism committed:
Avoid unnecessary +0 in transform-es2015-parameters (#4738)

motiz88 committed:
Merge pull request #4744 from nhajidin/fix-checkout

Fix line endings on checkout

SimenB committed:
Replace `path-exists` with `fs.existsSync` (#4731)

It's not deprecated anymore, see nodejs/node#8364

danez committed:
Fix replacing for-of if inside label (#4736)

This replaces the label instead of the for-of itself as we already
integrate the label in the replacement nodes.
Fixes #3858

motiz88 committed:
Support ObjectExpression in static path evaluation (#4746)

hzoo committed:
fixes [skip ci]

Ref https://github.com/babel/babel/issues/4590#issuecomment-254359917

garyjN7 committed:
Code review comment changes.

garyjN7 committed:
Move NODE_ENV variable setting.

garyjN7 committed:
Remove uneccessary on exit function

garyjN7 committed:
Removed files that are no longer used.

garyjN7 committed:
Updated documentation to use new npm build and test commands.

garyjN7 committed:
Merge remote-tracking branch 'upstream/master'
